modules/exploits/multi/http/atlassian_confluence_rce_cve_2023_22515.rb

```diff
@@ -0,0 +1,319 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Retry
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Atlassian Confluence Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an improper input validation issue in Atlassian Confluence, allowing arbitrary HTTP
+          parameters to be translated into getter/setter sequences via the XWorks2 middleware and in turn allows for
+          Java objects to be modified at run time. The exploit will create a new administrator user and  upload a
+          malicious plugins to get arbitrary code execution. All versions of Confluence between 8.0.0 through to 8.3.2,
+          8.4.0 through to 8.4.2, and 8.5.0 through to 8.5.1 are affected.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # MSF Exploit & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2023-22515'],
+          ['URL', 'https://attackerkb.com/topics/Q5f0ItSzw5/cve-2023-22515/rapid7-analysis'],
+          ['URL', 'https://confluence.atlassian.com/security/cve-2023-22515-privilege-escalation-vulnerability-in-confluence-data-center-and-server-1295682276.html'],
+        ],
+        'DisclosureDate' => '2023-10-04',
+        'Privileged' => false, # `NT AUTHORITY\NETWORK SERVICE` on Windows by default.
+        'Targets' => [
+          [
+            'Automatic',
+            {
+              'Platform' => 'java',
+              'Arch' => [ARCH_JAVA]
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          # Note we cannot delete the admin user we create, as Confluence prevents a user deleting themself.
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        # By default Confluence listens for HTTP requests on TCP port 8090.
+        Opt::RPORT(8090),
+        # Confluence may have a non default base path, allow user to configure that here.
+        OptString.new('TARGETURI', [true, 'Base path for Confluence', '/']),
+        # The endpoint we target to trigger the vulnerability.
+        OptString.new('CONFLUENCE_TARGET_ENDPOINT', [true, 'The endpoint used to trigger the vulnerability.', 'server-info.action']),
+        # We upload a new plugin, we need to wait for the plugin to be installed. This options governs how long we wait.
+        OptInt.new('CONFLUENCE_PLUGIN_TIMEOUT', [true, 'The timeout to wait when installing a plugin', 30])
```

Comment:
What's the unit for this number? Seconds, minutes, hours?

Reply:
resolved via [3242a70](https://github.com/rapid7/metasploit-framework/pull/18461/commits/3242a7009b477f0be030b880ba84cc0cf5fc9b7d)